src/Microsoft.Health.Dicom.Blob/Features/Export/AzureBlobExportSinkProvider.cs

```diff
@@ -125,6 +134,33 @@ protected override async Task<AzureBlobExportOptions> SecureSensitiveInfoAsync(A
     protected override Task ValidateAsync(AzureBlobExportOptions options, CancellationToken cancellationToken = default)
     {
         List<ValidationResult> results = options.Validate(new ValidationContext(this)).ToList();
+
+        // Run additional validation based on the sink provider options if the options themselves are configured correctly.
+        // We need to ensure that they do not contain SAS Tokens or use public access if it is disallowed
+        if (results.Count == 0)
+        {
+            if (options.BlobContainerUri != null)
+            {
+                if (options.BlobContainerUri.Query.Length > 1)
+                {
+                    if (!_providerOptions.AllowSasTokens)
+                        results.Add(new ValidationResult(DicomBlobResource.SasTokenAuthenticationUnsupported));
+                }
+                else if (!options.UseManagedIdentity && !_providerOptions.AllowPublicAccess)
+                {
+                    results.Add(new ValidationResult(DicomBlobResource.PublicBlobStorageConnectionUnsupported));
+                }
+            }
+            else if (!_providerOptions.AllowSasTokens && !options.IsEmulatorConnectionString() && options.ConnectionString.Contains("SharedAccessSignature=", StringComparison.OrdinalIgnoreCase))
```

Comment:
Do we care if they provide a `ConnectionString` and a `BlobContainerUri`? Is this just checked somewhere else?

Reply:
Yes! We check both